File: test/lang/1-fortran/fib_2.f90
module fibonacci_module
   use iso_fortran_env
   use, intrinsic :: iso_c_binding, only: c_long
   implicit none
   integer, parameter :: version = 2

   private
   public :: version, fib, printfib

contains

   recursive function fibalgo(value) result(n)
      integer(c_long), intent(in) :: value
      integer(c_long) :: i, l, n, p

      l = 0
      p = 1
      n = value
      
      do i = 1, value-1
         n = l + p
         l = p
         p = n
      end do
   end function fibalgo

   function fib(value) result(res) bind(c, name="fib")
      integer(c_long), intent(in) :: value
      integer(c_long) :: res
      res = fibalgo(value)
   end function fib

   subroutine printfib(value) bind(c, name="printfib")
      integer(c_long),value :: value
      write(*,99) "[Fortran Fibonacci Library v", version, "] fib(", value, ") = ", fibalgo(value)
   99 format(A, I0, A, I0, A, I0) 
      call flush(output_unit)
   end subroutine printfib

end module fibonacci_module

File: test/lang/1-c/fib_2.c
#include "fib.h"

#include <stdio.h>

const unsigned short version = 2;

static long fibalgo(long value) {
	long l = 0;
	long p = 1;
	long n = value;
	for (long i = 1; i < value; i++) {
		n = l + p;
		l = p;
		p = n;
	}
	return n;
}

long fib(long value) {
	return fibalgo(value);
}

void printfib(long value) {
	printf("[C Fibonacci Library v%d] fib(%ld) = %ld\n", version, value, fibalgo(value));
	fflush(stdout);
}